end_to_end_model_benchmark/module_inference.py

```diff
@@ -0,0 +1,128 @@
+logging.basicConfig(level=logging.INFO)
+
+import numpy as np
+import mxnet as mx
+import cv2
+from collections import namedtuple
+
+from collections import namedtuple
+Batch = namedtuple('Batch', ['data'])
+
+class InferenceHelper(object):
+    def __init__(self, opt):
+        # Pre-processing details
+        self.input_shape = (224, 224)
+        self.norm_mean = (0.485, 0.456, 0.406)
+        self.norm_std = (0.229, 0.224, 0.225)
+
+        self.model_path = opt.model_path
+        self.model_name = opt.model_name
+        self.iterations = opt.iterations
+        self.is_end_to_end = opt.end_to_end.lower() == 'true'
+        self.batch_size = opt.batch_size
+
+        self.ctx = mx.gpu(0) if int(opt.use_gpus) > 0 else mx.cpu()
+
+        # If end to end model, pre-processing is part of the network,
+        # we take raw input data in channels last format.
+        if self.is_end_to_end:
+            self.input_data_shape = (self.batch_size, 224, 224, 3)
+        else:
+            self.input_data_shape = (self.batch_size, 3, 224, 224)
+        
+        print("Running end-to-end v/s non-end-to-end inference benchmarks with below configuration: ")
+        print("Model Path - {}".format(self.model_path))
+        print("Model Name - {}".format(self.model_name))
+        print("Context - {}".format(self.ctx))
+        print("Is end to end model - {}".format(self.is_end_to_end))
+        print("Input shape - {}".format(self.input_data_shape))
+        print("Inference batch size - {}".format(self.batch_size))
+        print("Iterations - {}".format(self.iterations))
+
+    def download_model(self):
+        model_json_path = "{}/{}-symbol.json".format(self.model_path, self.model_name)
+        model_params_path = "{}/{}-0000.params".format(self.model_path, self.model_name)
+        print("Downloading the following model files...")
+        print(model_json_path)
+        print(model_params_path)
+
+        try:
+            mx.test_utils.download(model_json_path)
+            mx.test_utils.download(model_params_path)
+        except Exception as e:
+            print("ERROR: Failed to download the models. {}".format(e))
+    
+    def load_model(self):
+        sym, arg_params, aux_params = mx.model.load_checkpoint(self.model_name, 0)
+        self.mod = mx.mod.Module(symbol=sym, context=self.ctx, label_names=None)
+        self.mod.bind(for_training=False, data_shapes=[('data', self.input_data_shape)],
+                      label_shapes=self.mod._label_shapes)
+        self.mod.set_params(arg_params, aux_params, allow_missing=True)
+
+    def __preprocess_data(self, raw_data):
+        """Data pre-processing. Process 1 sample at a time.
+        Used in non end to end model benchmarking.
+        """
+        # Resize -> ToTensor -> Normalize
+        processed_data = [] #nd.array(shape = data.shape[0])
+        for i in range(raw_data.shape[0]):
+            img = mx.image.imresize(raw_data[i], self.input_shape[0], self.input_shape[1])
```

Comment:
nit:
imresize(data, *w*, *h*)
```suggestion
            img = mx.image.imresize(raw_data[i], self.input_shape[1], self.input_shape[0])
```

Reply:
Input shape in the constructor is in w,h format
self.input_shape = (224, 224)